ietf-layer0-types-ext.yang

```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
+    base modulation;
+    description
+    "DC DP-QAM16 (Dual Polarization Quadrature 
+    Amplitude Modulation)";
+  } 
+
+  identity FEC {
+    description
+      "Enumeration that defines the type of
+       Forward Error Correction";
+  }
+  identity reed-solomon {
+    base FEC;
+      description
+  "Reed-Solomon error correction";
+  }
+  identity hamming-code {
+    base FEC;
+      description
+  "Hamming Code error correction";
+  }
+  identity golay {
+    base FEC;
+      description "Golay error correction";
+  } 
+
+  identity line-coding {
+    description
+      "base line-coding class";
+    reference 
+      "ITU-T G.698.2-201811 section 7"   
+  }  
+
+  identity line-coding-NRZ-2p5G {
+    base line-coding;
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+
+  }
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+
+  }
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
+
+// temporary defined here for disalignment with 
+//ietf-layer0-types module
+  typedef standard-mode {
+    type string;
+    description
+      "ITU-T G.698.2 standard mode that guarantees 
+       interoperability.
+       It must be an string with the following format:
+       B-DScW-ytz(v) where all these attributes 
+       are conformant
+       to the ITU-T recomendation";
+    reference "ITU-T G.698.2 (11/2018)";
+  }
+
+  typedef frequency-thz {
+      type decimal64 {
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+    }
+
+// GROUPINGS  
+
+/* This grouping represent the list of capabilities attributes not under operational-mode */
+
+
+  grouping common-tranceiver-capabilities {
+    description "Capability of an optical tranceiver";
+
+ 
+/* transmitter tuning range (f_tx-min, f_tx-max) */
+
+    leaf min-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description
+        "his parameter indicates the minimum frequency for
+                   this template.";
+    }
+    leaf max-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description "This parameter indicates the minimum frequency
+                   for this template";
+    }
+
+    transmitter-tunability-grid {
+      type identityref {
+        base l0-types:flexi-ch-spc-type
+      }

```

Comment:
In order to allow the OTSi's central frequency not to stay on the G.694.1 grid, the type of this attribute should be a frequency in THz or GHz

Reply:
**2020-03-20 CCAMP call**

Should be:
```
            type decimal64 {
              fraction-digits 3;
            }
            units GHz;

```
It may be worthwhile defining a common type for this parameters, such as in https://tools.ietf.org/html/draft-ietf-ccamp-layer0-types-03:

```
    typedef frequency-ghz {
      type decimal64 {
        fraction-digits 3;
      }
      units GHz;
      description
        "The DWDM frequency in GHz, e.g., 193112.500";
      reference
        "RFC6205: Generalized Labels for
        Lambda-Switch-Capable (LSC) Label Switching Routers";
    }

```